pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Schema.java

```diff
@@ -213,6 +213,16 @@ default void configureSchemaInfo(String topic, String componentName,
         return DefaultImplementation.newProtobufSchema(SchemaDefinition.builder().withPojo(clazz).build());
     }
 
+    /**
+     * Disable SchemaInfo.
+     *
+     * @param schema instance
+     * @return a Schema instance of bytes type
+     */
+    static <T> Schema<T> disabled(Schema<T> schema) {
```

Comment:
Using the proposed API of wrapping the schema seems a bit weird to me:

```java
client.newProducer(Schema.disabled(Schema.JSON.of(MyClass.class))
    .topic("my-topic")
    .create();
```

In Pulsar the schema definition is used in 3 ways: 
 * Typed client API
 * SerDe
 * Server side schema validation

Using `disabled` doesn't convey which of these 3 aspect is being disabled.

What about having an option in the producer, like: 

```java
client.newProducer(Schema.JSON.of(MyClass.class)
    .topic("my-topic")
    .schemaDisableValidation(true) // ... or similar
    .create();
```

Reply:
@merlimat this is not about disabling schema validation. this is about disabling carrying schema information. this should be applied to both producer and consumers. hence I think it should be a schema wrapper over existing schema to turn off carrying schema information. We can choose a better name if `SchemaDisabled` is not accurate. but I don't think it should be added to producer or consumer configuration.

Reply:
>  this is not about disabling schema validation. this is about disabling carrying schema information

That was the point that "disabled" is not clear enough to convey what's being disabled.

> We can choose a better name if SchemaDisabled is not accurate. but I don't think it should be added to producer or consumer configuration.

Why not? To me it seems more of a producer/consumer property: the "schema" is the same, but you don't want to pass the schema info to broker.